### DISCUSSION TITLE
how to use taichi to decorate Class 怎么修饰类

### DISCUSSION THREAD

**Discussion by DaBaiTuuu** [taichi-dev/taichi#6096] [Q&A] [answered]
如题，这么使用 @ti.kernel 和@ti.func 修饰类或者类中函数

how to use taichi  @ti.kernel and @ti.func  to decorate Class AND function defined in a Class

**Comment by strongoier** [comment, marked as answer]
Hi @DaBaiTuuu. You may take a look at https://docs.taichi-lang.org/docs/odop.

**Reply by DaBaiTuuu** [reply]
aha!
Ok,thx .Cuz i did not find it last week, I thought the website may be updated, or i simply missed it.
Anyway thanks.
